packages/live-preview-vue/src/index.ts

```diff
@@ -0,0 +1,45 @@
+import { ready, subscribe, unsubscribe } from '@payloadcms/live-preview'
+import { type Ref, ref, watchEffect } from 'vue'
+
+/**
+ * Vue composable to implement Payload CMS Live Preview.
+ *
+ * {@link https://payloadcms.com/docs/live-preview/frontend View the documentation}
+ */
+export const useLivePreview = <T>(props: {
+  apiRoute?: string
+  depth?: number
+  initialData: T
+  serverURL: string
+}): { data: Ref<T>; isLoading: Ref<boolean> } => {
+  const { apiRoute, depth, initialData, serverURL } = props
+  const data = ref(initialData) as Ref<T> // Workaround for weird Ref<T> behavior
+  const isLoading = ref(true)
+  const hasSentReadyMessage = ref(false)
+
+  const onChange = (mergedData: T) => {
+    data.value = mergedData
+    isLoading.value = false
+  }
+
+  watchEffect((onCleanup) => {
```

Comment:
I think it is change to `onMounted`. `useEffect` is not call on server, but watchEffect is called in server.

Reply:
@SimYunSup is this what you had in mind? [`750a8d5` (#4987)](https://github.com/payloadcms/payload/pull/4987/commits/750a8d53ee11ffc52448abdd9a0f380c7ff8e53f)

Also @ChrisGV04 can you test the change to be sure it still works?

Reply:
Good observation @SimYunSup!

@DanRibbens I tested the changes on both regular Vue 3 and Nuxt 3 (with SSR). It works as intended and with SSR the subscription only runs on the client.

Reply:
`onUnmounted` is not cleanup function. I think it will be changed like `@xstate/vue` ([code](https://github.com/statelyai/xstate/blob/54a2b01e666a705c643ff7b1a7b0bf7a074aa9c1/packages/xstate-vue/src/useActorRef.ts#L22-L33)).

```ts
let subscription;
onMounted(() => {
  subscription = subscribe()
})
onBeforeUnmount(() => {
  unsubscribe(subscription);
});
```

Reply:
[The Vue docs](https://vuejs.org/api/composition-api-lifecycle.html#onunmounted) themselves recommend using `onUnmounted` to cleanup subscriptions. Even VueUse that's made by the Vue core team uses it to cleanup watchers and timers.

Maybe it's better to move it out of `onMounted` but even like this it works exactly as intended.